src/components/Fieldset/DynamicFieldset/ControlledDynamicFieldset.tsx

```diff
@@ -4,42 +4,41 @@ import '../Fieldset.scss';
 import type {ControlledDynamicFieldsetProps} from './DynamicFieldset.types';
 import {Switch, Typography} from '~/components';
 
-export const ControlledDynamicFieldset = React.forwardRef<HTMLDivElement, ControlledDynamicFieldsetProps>(({
+export const ControlledDynamicFieldset = React.forwardRef<HTMLFieldSetElement, ControlledDynamicFieldsetProps>(({
     id,
     label,
     helper,
     children,
     className,
     buttons,
-    isOpen = false,
+    checked = false,
     onChange,
     ...props
 }, ref) => {
     return (
-        <div
+        <fieldset
             ref={ref}
             id={id}
             className={clsx(
                 'moonstone-dynamic-fieldset',
-                isOpen && 'moonstone-dynamic-fieldset_open',
+                checked && 'moonstone-dynamic-fieldset_open',
                 'flexCol_nowrap',
                 className
             )}
+            aria-checked={checked}
             {...props}
         >
-            <div className={clsx('flexRow_nowrap', 'flexFluid', 'alignCenter')}>
-                <div className="flexRow_nowrap flexFluid alignCenter">
-                    <Typography isNowrap component="label" variant="heading" weight="bold">{label}</Typography>
-                </div>
-                <Switch checked={isOpen} onChange={onChange}/>
+            <legend className={clsx('flexRow_nowrap', 'flexFluid', 'alignCenter')}>
+                <Typography isNowrap className="flexRow_nowrap flexFluid alignCenter" component="label" variant="heading" weight="bold">{label}</Typography>
+                <Switch id="moonstone-dynamic-fieldset-switch" checked={checked} onChange={onChange}/>
                 {buttons && buttons}
-            </div>
+            </legend>
             {helper &&
                 <Typography variant="caption" className={clsx('moonstone-fieldset_helper')}>{helper}</Typography>}
             <div className={clsx('moonstone-fieldset_children', 'flexCol_nowrap')}>
```

Comment:
The condition `checked && children` should also wrap the `div`, to avoid having an empty `div` when no `children` is provided or when when the fieldset is collapsed.